calour/io.py

```diff
@@ -103,23 +144,58 @@ def _read_qiime2(fp, transpose=True):
     feature_md : pandas.DataFrame
         the feature metadata (if availble in table)
     '''
-    import qiime2
-    logger.debug('loading qiime2 biom table %s' % fp)
+    logger.debug('loading qiime2 feature table %s' % fp)
 
-    q2table = qiime2.Artifact.load(fp)
-    table = q2table.view(biom.Table)
+    # load the feature table file
+    with tempfile.TemporaryDirectory() as tempdir:
+        oname = _file_from_zip(tempdir, fp, internal_data='data/feature-table.biom')
+        sid, fid, data, fmd = _read_biom(oname, transpose=transpose)
 
-    sid = table.ids(axis='sample')
-    fid = table.ids(axis='observation')
-    logger.info('loaded %d samples, %d observations' % (len(sid), len(fid)))
-    data = table.matrix_data
-    feature_md = _get_md_from_biom(table)
+    return sid, fid, data, fmd
 
-    if transpose:
-        logger.debug('transposing table')
-        data = data.transpose()
 
-    return sid, fid, data, feature_md
+def read_qiime2(fp, sample_metadata_file=None, rep_seq_file=None, taxonomy_file=None, use_qiime2=False, **kwargs):
+    '''Read a qiime2 feature table and additional optional artifact files (representative sequences and taxonomy) into a Calour.AmpliconExperiment
+
+    Parameters
+    ----------
+    fp: str
+        name of the qiime2 feature table .qza artifact file
+    sample_metadata_file : None or str, optional
+        None (default) to just use sample names (no additional metadata).
+        if not None, file path to the sample metadata (aka mapping file in QIIME).
+    rep_seq_file: None or str, optional
+        None (default) to use the feature ids in the feature table
+        if not None, file path to the qiime2 representative sequences artifact file (defined by the qiime2 --o-representative-sequences parameter)
+    taxonomy_file: None or str, optional
+        if not None, add taxonomy for each feature using the qiime2 taxonomy artifact file (output of the qiime2 feature-classifier command)
+
+    Keyword Arguments
+    -----------------
+    %(io.read.parameters)s
+    '''
+    newexp = read_amplicon(fp, sample_metadata_file=sample_metadata_file, data_file_type='qiime2', **kwargs)
+    with tempfile.TemporaryDirectory() as tempdir:
```

Comment:
if using zipfile.open, we don't need temp dir?

Reply:
addressed in prev. comment. only works in python 3.7